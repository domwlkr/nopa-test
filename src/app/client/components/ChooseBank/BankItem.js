import React, { PropTypes} from 'react';

const BankItem = (props) => {
  return (
    <li className={"bank-item " + (props.isSelected ? 'selected' : '')} onClick={() => props.onSelectBank(props.bank)}>
      <img className="bank-image" alt={props.bank.name} src={props.bank.logo} />
    </li>
  );
};

BankItem.proptypes = {
  bank: PropTypes.object.isRequired,
  onSelectBank: PropTypes.func.isRequired
};

export default BankItem;