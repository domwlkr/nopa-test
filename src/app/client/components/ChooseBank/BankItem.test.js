import React from 'react';
import BankItem from './BankItem';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const bank = {
  name: 'Barclays',
  logo: 'test-file-stub'
};

const onSelectBank = jest.fn();

test('Renders correctly', () => {
  const bankItem = renderer.create(
    <BankItem bank={bank} onSelectBank={onSelectBank} />
  ).toJSON();

  expect(bankItem).toMatchSnapshot();
});

test('Displays the correct bank', () => {
  const bankItem = shallow(
    <BankItem bank={bank} onSelectBank={onSelectBank} />
  );

  expect(bankItem.instance().props.bank.name).toEqual('Barclays');
});

test('Item click event fired', () => {
  const bankItem = shallow(
    <BankItem bank={bank} onSelectBank={onSelectBank} />
  );

  bankItem.simulate('click');

  expect(onSelectBank).toHaveBeenCalled();
});