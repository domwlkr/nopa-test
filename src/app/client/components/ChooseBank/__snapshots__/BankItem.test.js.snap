// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<li
  className="bank-item "
  onClick={[Function]}
>
  <img
    alt="Barclays"
    className="bank-image"
    src="test-file-stub"
  />
</li>
`;
